### PR TITLE
Cleanup mime attribute of forceimprovedsupportcheck

### PR DESCRIPTION
**Pull request by borongc**
Remove this attribute as it is not used.

b/216483296

### CODE REVIEW

File: starboard/android/shared/media_capabilities_cache.cc [hunk 1]
```diff
@@ -231,23 +231,9 @@ bool VideoCodecCapability::IsBitrateSupported(int bitrate) const {
   return supported_bitrates_.Contains(bitrate);
 }
 
-bool VideoCodecCapability::AreResolutionAndRateSupported(
-    bool force_improved_support_check,
-    int frame_width,
-    int frame_height,
-    int fps) {
-  if (force_improved_support_check) {
```
Comment (osagie98) [inline starboard/android/shared/media_capabilities_cache.cc:239]: `force_improved_support_check` is on by default (check line 87 in media_is_video_supported.cc). If we're removing it, we still need to keep this logic to maintain the default behavior.

File: starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaCodecUtil.java [hunk 2]
```diff
@@ -659,8 +656,7 @@ public static String findVideoDecoder(
       // Enable the improved support check based on more specific APIs, like isSizeSupported() or
       // areSizeAndRateSupported(), for 8k content. These APIs are theoretically more accurate,
       // but we are unsure about their level of support on various Android TV platforms.
-      final boolean enableImprovedSupportCheck =
```
Comment (osagie98) [inline starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaCodecUtil.java:662]: Like below, `force_improved_support_check` is on by default, so |enableImprovedSupportCheck| should probably be true by default now, or removed.

File: starboard/android/shared/media_capabilities_cache.cc (hunk 1 above)
Comment (borongc) [reply starboard/android/shared/media_capabilities_cache.cc:239]: Remove `force_improved_support_check ` and keep the logic inside the statement.

File: starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaCodecUtil.java (hunk 2 above)
Comment (borongc) [reply starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaCodecUtil.java:662]: Only remove `enableImprovedSupportCheck `.

File: starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaCodecUtil.java [hunk 3]
```diff
@@ -659,35 +656,20 @@ public static String findVideoDecoder(
       // Enable the improved support check based on more specific APIs, like isSizeSupported() or
       // areSizeAndRateSupported(), for 8k content. These APIs are theoretically more accurate,
```
Comment (osagie98) [inline starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaCodecUtil.java:657]: You can remove this comment now since the boolean is gone.

Comment (borongc) [reply starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaCodecUtil.java]: Done.

File: starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaCodecUtil.java [hunk 4]
```diff
@@ -552,8 +551,7 @@ public static String findVideoDecoder(
             Locale.US,
             "Searching for video decoder with parameters mimeType: %s, secure: %b, frameWidth:"
                 + " %d, frameHeight: %d, bitrate: %d, fps: %d, mustSupportHdr: %b,"
-                + " mustSupportSoftwareCodec: %b, mustSupportTunnelMode: %b,"
-                + " forceImprovedSupportCheck: %b",
+                + " mustSupportSoftwareCodec: %b, mustSupportTunnelMode: %b,",
```
Comment (osagie98) [inline starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaCodecUtil.java:554]: nit: remove the comma after "mustSupportTunnelMode: %b,"

**Review by osagie98 — APPROVED** [review]

**Pull request merged**